pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -0,0 +1,2272 @@
+        results = mipopt.solve(m, 
+                            tee=config.mip_solver_tee, 
+                            load_solutions=False,
+                            **mip_args)
+        if len(results.solution) > 0:
+            m.solutions.load_from(results)
+
+        solve_terminate_cond = results.solver.termination_condition
+        if solve_terminate_cond is tc.optimal:
+            copy_var_list_values(
+                MindtPy.variable_list,
+                self.working_model.MindtPy_utils.variable_list,
+                config)
+            config.logger.info(self.log_formatter.format('-',
+                                                            'Max binary MILP', 
+                                                            value(MindtPy.max_binary_obj.expr),
+                                                            self.primal_bound,
+                                                            self.dual_bound,
+                                                            self.rel_gap,
+                                                            get_main_elapsed_time(self.timing)))
+        elif solve_terminate_cond is tc.infeasible:
+            raise ValueError(
+                'MILP main problem is infeasible. '
+                'Problem may have no more feasible '
+                'binary configurations.')
+        elif solve_terminate_cond is tc.maxTimeLimit:
+            config.logger.info(
+                'NLP subproblem failed to converge within time limit.')
+            self.results.solver.termination_condition = tc.maxTimeLimit
+        elif solve_terminate_cond is tc.maxIterations:
+            config.logger.info(
+                'NLP subproblem failed to converge within iteration limit.')
+        else:
+            raise ValueError(
+                'MindtPy unable to handle MILP main termination condition '
+                'of %s. Solver message: %s' %
+                (solve_terminate_cond, results.solver.message))
+
+    ##################################################################################################################################################################################################################
+    # nlp_solve.py
+
+    def solve_subproblem(self, config):
+        """Solves the Fixed-NLP (with fixed integers).
+
+        This function sets up the 'fixed_nlp' by fixing binaries, sets continuous variables to their intial var values,
+        precomputes dual values, deactivates trivial constraints, and then solves NLP model.
+
+        Parameters
+        ----------
+        config : ConfigBlock
+            The specific configurations for MindtPy.
+
+        Returns
+        -------
+        fixed_nlp : Pyomo model
+            Integer-variable-fixed NLP model.
+        results : SolverResults
+            Results from solving the Fixed-NLP.
+        """
+        fixed_nlp = self.working_model.clone()
```

Comment:
Oh. Cloning the model is **slow**? Do you know how much performance will be improved if we replace cloning the model?

Reply:
Currently, we also clone the model when an incumbent solution is found. Will it be faster if we only copy the values instead of cloning the model?

https://github.com/ZedongPeng/pyomo/blob/mindtpy-rewrite/pyomo/contrib/mindtpy/algorithm_base_class.py#L990-L991

Reply:
@jsiirola Can you let us know if this would every affect the performance?

Reply:
My guess is that most of mindtpy's runtime is usually spent inside of a solver... If that's the case, then this probably isn't the bottleneck (and either way, I don't think you should address this in this PR). But it is true that it will be faster to transfer solution information and make some modifications to the model state than cloning a couple times per iteration. What you do now though, is probably the safest in terms of avoiding bugs. Because you don't have to worry about things like undoing preprocessing that was only valid for a specific iteration, for example.